spark-common/src/main/java/me/lucko/spark/common/sampler/java/TickedDataAggregator.java

```diff
@@ -151,6 +151,17 @@ public List<ThreadInfo> getList() {
             return this.list;
         }
 
+        public int sizeWithoutTrailingSleeping() {
+            // find the last index at which the thread wasn't sleeping
+            int i;
+            for (i = this.list.size() - 1; i >= 0; i--) {
+                if (!isSleeping(this.list.get(i))) {
+                    return i;
```

Comment:
I think this should probably return `i + 1` because initially `i` is set to `this.list.size() - 1`?

Reply:
Fixed